### PR TITLE
feat: Dbtp 1162 support hosting static sites on s3

### PR DESCRIPTION
**Pull request by gabelton**
https://uktrade.atlassian.net/browse/DBTP-1162

### CODE REVIEW

File: s3/main.tf [hunk 1]
```diff
@@ -149,6 +161,223 @@ resource "aws_s3_bucket_public_access_block" "public_access_block" {
...
+data "aws_cloudfront_cache_policy" "example" {
+  name = "Managed-CachingOptimized"
+}
+
+resource "aws_cloudfront_distribution" "s3_distribution" {
+  # checkov:skip=CKV2_AWS_32: Ensure CloudFront distribution has a response headers policy attached
+  # checkov:skip=CKV2_AWS_47: Ensure AWS CloudFront attached WAFv2 WebACL is configured with AMR for Log4j Vulnerability
+  # checkov:skip=CKV_AWS_68: CloudFront Distribution should have WAF enabled
+  # checkov:skip=CKV_AWS_86: Ensure CloudFront distribution has Access Logging enabled
+  # checkov:skip=CKV_AWS_305: Ensure CloudFront distribution has a default root object configured
```
Comment (gabelton) [inline s3/main.tf:287]: I don't think we want this, as we'll want individual service teams to configure whatever objects they want

File: s3/main.tf [hunk 2]
```diff
@@ -149,6 +161,223 @@ resource "aws_s3_bucket_public_access_block" "public_access_block" {
...
+  }
+}
+
+data "aws_cloudfront_cache_policy" "example" {
+  name = "Managed-CachingOptimized"
+}
+
+resource "aws_cloudfront_distribution" "s3_distribution" {
+  # checkov:skip=CKV2_AWS_32: Ensure CloudFront distribution has a response headers policy attached
+  # checkov:skip=CKV2_AWS_47: Ensure AWS CloudFront attached WAFv2 WebACL is configured with AMR for Log4j Vulnerability
```
[the remark this answers is not in the document]
Comment (ejayesh) [reply s3/main.tf:284]: You are correct, it is not needed in this case.

File: s3/main.tf (hunk 1 above)
Comment (ejayesh) [reply s3/main.tf:287]: Correct, this needs to be available to the team to configure.

File: s3/main.tf [hunk 3]
```diff
@@ -149,6 +161,223 @@ resource "aws_s3_bucket_public_access_block" "public_access_block" {
...
+  # checkov:skip=CKV_AWS_68: CloudFront Distribution should have WAF enabled
+  # checkov:skip=CKV_AWS_86: Ensure CloudFront distribution has Access Logging enabled
+  # checkov:skip=CKV_AWS_305: Ensure CloudFront distribution has a default root object configured
+  # checkov:skip=CKV_AWS_310: Ensure CloudFront distributions should have origin failover configured
```
[the remark this answers is not in the document]
Comment (ejayesh) [reply s3/main.tf:288]: We do not need to add this now

**Review by ejayesh — COMMENTED** [review]
@gabelton before approving, I would like to run deploy the terraform myself.  As its a complex change.

**Review by ejayesh — APPROVED** [review]
I am happy with all the changes now.  Approving

**Pull request merged**